### DISCUSSION TITLE
Add dynamic casts

### DISCUSSION THREAD

**Discussion by jaulz** [spatie/laravel-data#302] [Q&A]
Hey folks,

is there an easy way to add casts dynamically? I implemented a custom cast which receives a parameter and this parameter should be dependent on an input parameter of the `from` method. As far I could read the code, casts can only be added via attributes and these are per definition static but is there maybe any workaround?

Thanks!

**Comment by rubenvanassche** [comment]
At the moment this is not possible and I don't think we want to add something like that since it probably is going to open a box of pandora 😄.

At the moment a cast receives a cast context which are all the properties tied to the data object. Maybe we should add an extra `WithCreationalArgumentsCast` interface which injects the parameters.

But not entirely sure how to do that clean since at the moment we're throwing away anything provided to from as soon as the pipeline is started.
